searchresult.py

```diff
@@ -42,6 +44,10 @@ def are_identical(self, root1, root2) -> bool:
             return False
         if not root1.children and not root2.children and root1.text != root2.text:
             return False
+        text_comparison = root1.text == root2.text if not self.case_insensitive else root1.text.lower() == root2.text.lower()
```

Comment:
Have you considered a text_comparison function to hide the implementation details?